File: backend/src/main/java/com/godeltech/mastery/backend/service/BrandService.java
package com.godeltech.mastery.backend.service;

import com.godeltech.mastery.backend.domain.dto.responce.BrandDTO;
import com.godeltech.mastery.backend.domain.dto.responce.ModelDTO;
import com.godeltech.mastery.backend.domain.entity.Brand;
import java.util.List;

public interface BrandService {

  Brand getById(Long id);

  List<BrandDTO> getAllBrands();

  List<ModelDTO> getModelsByBrandId(Long id);
}

File: backend/src/main/kotlin/com/godeltech/mastery/backend/service/BrandService.kt
package com.godeltech.mastery.backend.service

import com.godeltech.mastery.backend.domain.dto.responce.BrandDto
import com.godeltech.mastery.backend.domain.dto.responce.ModelDto
import com.godeltech.mastery.backend.domain.entity.Brand

interface BrandService {

    fun getById(id: Long): Brand

    fun getAllBrands(): MutableList<BrandDto>

    fun getModelsByBrandId(id: Long): MutableList<ModelDto>
}